front/src/components/CardList.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <div class="card-list">
+    <div
+      v-if="loading"
+      class="is-flex is-justify-content-center is-align-items-center my-6"
+    >
+      <div class="loader"></div>
+    </div>
+    <div
+      v-if="!loading && error"
+      class="is-flex is-justify-content-center is-align-items-center my-6"
+    >
+      <p>
+        Sorry there was an error. Please try again or contact tha IT
```

Comment:
Maybe just "Sorry there was an error. You may find more details in your Javascript console, or in Showergel's logs"

because this will be self-hosted by the admin.